src/effects/queue.rs

```diff
@@ -0,0 +1,224 @@
+        true,
+    );
+
+    e.footer(|f| {
+        let seconds = total_duration.as_secs() % 60;
+        let minutes = total_duration.as_secs() / 60;
+
+        f.text(format!(
+            "Page {}/{} | Total Duration: {:02}:{:02}",
+            page + 1,
+            (queue.len() / 10 + 1),
+            minutes,
+            seconds,
+        ))
+    });
+
+    e
+}
+
+fn build_action_row(page: usize, queue_len: usize) -> CreateActionRow {
+    let pages = queue_len / 10;
+    let mut row = CreateActionRow::default();
+
+    // Previous button
+    if page > 0 {
+        row.create_button(|b| {
+            b.style(ButtonStyle::Primary);
+            b.label("Previous");
+            b.custom_id(PREV_ID);
+            b.disabled(false)
+        });
+    } else {
+        row.create_button(|b| {
+            b.style(ButtonStyle::Danger);
+            b.label("Previous");
+            b.custom_id(PREV_ID);
+            b.disabled(true)
+        });
+    }
+
+    // Next button
+    if pages >= 1 && page < pages {
+        row.create_button(|b| {
+            b.style(ButtonStyle::Primary);
+            b.label("Next");
+            b.custom_id(NEXT_ID);
+            b.disabled(false)
+        });
+    } else {
+        row.create_button(|b| {
+            b.style(ButtonStyle::Danger);
+            b.label("Next");
+            b.custom_id(NEXT_ID);
+            b.disabled(true)
+        });
+    }
+
+    row
+}
+
+async fn get_queue(ctx: &Context, guild_id: GuildId) -> SunnyResult<Vec<TrackHandle>> {
+    Ok(songbird::get(ctx)
+        .await
+        .ok_or_else(|| SunnyError::log("Couldn't get songbird"))?
+        .get(guild_id)
+        .ok_or_else(|| SunnyError::user("Not currently in a call"))?
+        .lock()
+        .await
+        .queue()
+        .current_queue())
+}
+
+/// Sends an interactive queue embed and interactions
+pub async fn send_embed(
+    ctx: &Context,
+    guild_id: GuildId,
+    channel_id: ChannelId,
+) -> SunnyResult<()> {
+    // Retrieve the current queue
+    let cq = get_queue(ctx, guild_id).await?;
+
+    // Send initial queue message
+    let message = channel_id
+        .send_message(&ctx.http, |m| {
+            m.components(|c| c.set_action_rows(vec![build_action_row(0, cq.len())]));
+            m.set_embed(generate_embed(&cq, 0))
+        })
+        .await
+        .map_err(|e| SunnyError::log(format!("Unable to send queue message: {:?}", e).as_str()))?;
+
+    await_interactions(ctx, message, guild_id).await
+}
+
+async fn await_interactions(
+    ctx: &Context,
+    mut message: Message,
+    guild_id: GuildId,
+) -> SunnyResult<()> {
+    // Currently shown page
+    let mut page: usize = 0;
+
+    // await interactions i.e. button presses
+    let mut collector = message
+        .await_component_interactions(&ctx.shard)
+        .timeout(Duration::from_secs(3600)) // 1h
+        .await;
+
+    // Process button presses
+    while let Some(mci) = collector.next().await {
```

Comment:
Should we live update the queue every `t`?

Reply:
How do you mean?

Reply:
Gonna leave this for now, per dm